libtk205/cmake/update_lib_repo.cmake

```diff
@@ -0,0 +1,117 @@
+                OUTPUT_VARIABLE last_commit_msg
+                OUTPUT_STRIP_TRAILING_WHITESPACE
+)
+message(STATUS "Last commit message on local branch was \"${last_commit_msg}\"")
+
+execute_process(COMMAND ${GIT_EXECUTABLE} ls-remote --heads --exit-code ${authenticated_repo} ${current_git_branch}
+                WORKING_DIRECTORY ${clone_dir}
+                RESULT_VARIABLE exit_code
+                OUTPUT_VARIABLE output_repo_branches
+)
+if(exit_code EQUAL "0")
+   message(STATUS "Branch ${current_git_branch} exists in upload remote.")
+   execute_process(COMMAND ${GIT_EXECUTABLE} checkout ${current_git_branch}
+                   WORKING_DIRECTORY ${clone_dir}
+   )
+else()
+   message(STATUS "Branch ${current_git_branch} must be created in upload remote.")
+   execute_process(COMMAND ${GIT_EXECUTABLE} checkout -b ${current_git_branch}
+                   WORKING_DIRECTORY ${clone_dir}
+   )
+   execute_process(COMMAND ${GIT_EXECUTABLE} push -u origin ${current_git_branch}
+                   WORKING_DIRECTORY ${clone_dir}
+   )
+endif()
+
+set(include_dest_dir "${clone_dir}/include/${repo_name}")
+set(src_dest_dir "${clone_dir}/src")
+
+file(MAKE_DIRECTORY "${include_dest_dir}")
+file(MAKE_DIRECTORY "${src_dest_dir}")
+
+# Copy generated files 
+file(COPY "${PROJECT_SOURCE_DIR}/../schema-205/build/include/"
+     DESTINATION "${include_dest_dir}"
+     FILES_MATCHING PATTERN *.h)
+file(COPY "${PROJECT_SOURCE_DIR}/include/"
+     DESTINATION "${include_dest_dir}"
+     FILES_MATCHING PATTERN *.h)
+file(COPY "${PROJECT_SOURCE_DIR}/../schema-205/schema205/src/"
+     DESTINATION "${include_dest_dir}"
+     FILES_MATCHING PATTERN *.h)
+
+file(COPY "${PROJECT_SOURCE_DIR}/../schema-205/build/cpp/"
+     DESTINATION "${src_dest_dir}"
+     FILES_MATCHING PATTERN *.cpp)
+file(COPY "${PROJECT_SOURCE_DIR}/src/"
+     DESTINATION "${src_dest_dir}"
+     FILES_MATCHING PATTERN *.cpp)
+file(COPY "${PROJECT_SOURCE_DIR}/../schema-205/schema205/src/"
+     DESTINATION "${src_dest_dir}"
+     FILES_MATCHING PATTERN *.cpp)
+
+file(COPY "${PROJECT_SOURCE_DIR}/test/"
+     DESTINATION "${clone_dir}/test"
+     FILES_MATCHING 
+      PATTERN *.cpp
+      PATTERN *.hpp)
+
+file(COPY "${PROJECT_SOURCE_DIR}/../schema-205/examples"
+     DESTINATION "${clone_dir}")
+
+# git config
+execute_process(COMMAND ${GIT_EXECUTABLE} config user.name "Actions CI"
+                WORKING_DIRECTORY ${clone_dir}
+)
+execute_process(COMMAND ${GIT_EXECUTABLE} config user.email "ci.bigladdersoftware.com"
+                WORKING_DIRECTORY ${clone_dir}
+)
+# git add
+execute_process(COMMAND ${GIT_EXECUTABLE} add include src test examples
+                WORKING_DIRECTORY ${clone_dir}
+)
+# git commit
+execute_process(COMMAND ${GIT_EXECUTABLE} commit -m "\"${last_commit_msg}\""
+                WORKING_DIRECTORY ${clone_dir}
+)
+# git push
+execute_process(COMMAND ${GIT_EXECUTABLE} push
+                WORKING_DIRECTORY ${clone_dir}
+)
```

Comment:
I was looking into whether we should port the 'vendor' subdirectory to the libtk205 repo as well as the source code, but my feeling now is that we should not. (libtk205's submodules are actually submodules of toolkit-205, so there wouldn't always be a 1:1 match.)